Tests/Realm.Tests/SourceGeneration/TestClasses/ManuallyGeneratedClass.cs

```diff
@@ -0,0 +1,270 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Realm.Generated;
+using Realms;
+using Realms.Schema;
+using Realms.Tests.SourceGeneration.TestClasses;
+using Realms.Weaving;
+
+namespace Realms.Tests.SourceGeneration.TestClasses
+{
+    public partial class ManualllyGeneratedClass : IRealmObject
+    {
+        #region Original class
+
+        //public int Integer { get; set; }
+
+        //public IList<int> IntegerList { get; }
+
+        //[MapTo("_name")]
+        //public string Name { get; set; }
+
+        //[PrimaryKey]
+        //public int PKey { get; set; }
+
+        #endregion
+
+        #region Weaved class
+
+        public int IntValue
+        {
+            get => _accessor.IntValue;
+            set => _accessor.IntValue = value;
+        }
+
+        public IList<int> ListValue => _accessor.ListValue;
+
+        [MapTo("_string")]
+        public string StringValue
+        {
+            get => _accessor.StringValue;
+            set => _accessor.StringValue = value;
+        }
+
+        [PrimaryKey]
+        public int PrimaryKeyValue
+        {
+            get => _accessor.PrimaryKeyValue;
+            set => _accessor.PrimaryKeyValue = value;
+        }
+
+        #endregion
+    }
+
+    [Woven(typeof(ManuallyGeneratedClassObjectHelper))]
+    public partial class ManualllyGeneratedClass : IRealmObject, INotifyPropertyChanged
+    {
+        // The schema property could be part of an interface, but for that the user needs to have at least .NET 5.0 and C# 8.0
+        // So we need to use reflection to check if this exists.
+        public static ObjectSchema RealmSchema = new ObjectSchema.Builder("ManualllyGeneratedClass", isEmbedded: false)
+        {
+            Property.Primitive("IntValue", RealmValueType.Int),
+            Property.PrimitiveList("ListValue", RealmValueType.Int),
+            Property.Primitive("_string", RealmValueType.String),
+            Property.Primitive("PrimaryKeyValue", RealmValueType.Int, isPrimaryKey: true),
+        }.Build();
+
+        #region IRealmObject implementation
+
+        private IManuallyGeneratedClassAccessor _accessor;
+
+        public IRealmAccessor Accessor => _accessor;
+
+        public bool IsManaged => _accessor.IsManaged;
+
+        public bool IsValid => _accessor.IsValid;
+
+        public bool IsFrozen => _accessor.IsFrozen;
+
+        public Realm Realm => _accessor.Realm;
+
+        public ObjectSchema ObjectSchema => _accessor.ObjectSchema;
```

Comment:
I wonder if exposing this on the user class is particularly useful. We may want to remove it from the interface and instruct people to use the accessor if they need it.